app/views/two_factor_authentications/_deactivate.haml

```diff
@@ -0,0 +1,28 @@
+.col-md-12
+  .row
+    .col-md-12
+      %h3= t("two_factor_auth.title")
+      %p= t("two_factor_auth.explanation")
+      .well= t("two_factor_auth.activated.status")
+
+      %hr
+
+      %h4= t("two_factor_auth.activated.change_button")
+      %p= t("two_factor_auth.activated.change_label")
+
+      = form_for "two_factor_authentication", url: two_factor_authentication_path,
+        html: {method: :delete, class: "form-horizontal"} do |f|
+        .form-group
+          = f.label :code, t("two_factor_auth.input_token.label"), class: "control-label col-sm-6"
+          .col-sm-6
+            = f.text_field :code, placeholder: t("two_factor_auth.input_token.placeholder"), class: "form-control"
+            = t("two_factor_auth.recovery.reminder")
```

Comment:
I tried to disable 2FA with a recovery code and it didn't work (the page reloaded but nothing changed). I know the recovery code is fine as I just used it to log in. Does that work for you?

Reply:
It did work for me, did you maybe use the same recovery code to login as to deactivate? It looks like they're one time use only: https://github.com/tinfoil/devise-two-factor/blob/master/lib/devise_two_factor/models/two_factor_backupable.rb#L40-L41

Reply:
I clarified they're single use